front-end/src/components/analysis-page/IntentMenu.test.js

```diff
@@ -120,6 +120,128 @@ describe("IntentMenu", () => {
+        });
+
+        it ("should correctly display 1 intent when only one is starred", () => {
+            // Create a deep copy to prevent change of star attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].star = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            // A button for Intent 0 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 1 ? 1 : 0);
+            });
+        });
+
+        it ("should correctly display 1 intent when only one is flagged", () => {
+            // Create a deep copy to prevent change of flag attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for Intent 0 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 1 ? 1 : 0);
+            });
+        });
+
+        it ("should correctly display 2 intents when two are starred", () => {
+            // Create a deep copy to prevent change of star attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].star = true;
+            props.intents[1].star = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+        });
+
+        it ("should correctly display 2 intents when two are flagged", () => {
+            // Create a deep copy to prevent change of flag attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].flag = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+        });
+
+        it ("should display only one intent when a single intent is both starred and flagged", () => {
+            // Create a deep copy to prevent change of attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
```

Comment:
I think a more efficient way of doing this would be to put it into a beforeEach and instead of making a deep copy, just reset star/flag to false in every intent before every test. This should be more efficient by a significant margin since parse/stringify are quite costly.